src/classes/PS_IntegrationServiceConfig.cls

```diff
@@ -66,12 +92,81 @@ public with sharing class PS_IntegrationServiceConfig {
     * @return Boolean
     */
     public Boolean hasIntegrationPermissions() {
-        // If the integration service is not enabled, user does not have the integration permissions
-        if (!isEnabled) {
-            return false;
+        Payment_Services_Configuration__c lastModifiedRecord = new Configuration().lastModifiedRecord;
+        return lastModifiedRecord != null 
+                ? lastModifiedRecord.LastModifiedById == UserInfo.getUserId() 
+                : false;
+    }
+
+
+
+    /**
+    * @description Inner class responsible for retrieving Payment Services configuration from
+    *              application storage and building and exposing a set of key-value
+    *              configuration pairs.
+    */
+    @TestVisible
+    private class Configuration {
```

Comment:
@vmartinov-sforg Since this class is potentially called by standard end users (via other logic), and the outer class uses `with sharing`, it might be helpful to declare this inner class as `without sharing` to ensure that the queries against the config objects always work.

Reply:
@force2b What would be the purpose to call this class for any other purpose? I'm reluctant to use `without sharing` and raise security concerns.

Reply:
My concern was that the class gets called from the trigger which is executed when a standard User makes a changes to the RD. If that standard user has no permissions to the `Payment_Services_Configuration__c` object, it is possible the query against that object could fail. Given the usage though, the intent is to determine if the current user is the integration, regardless of whether the current user actually has permissions to the `Payment_Services_Configuration__c` object. My proposal is to use `without sharing` only on the inner class to ensure that the query against `Payment_Services_Configuration__c` always works regardless of what permissions the current user has. It's also feasible to use a separate inner class that has `without sharing` just for the query and have that return the result back to this inner class. That might make it clearer for security review.